pkg/patterns/declarative/pkg/manifest/objects.go

```diff
@@ -305,6 +306,13 @@ func (o *Object) GroupVersionKind() schema.GroupVersionKind {
 	return o.object.GroupVersionKind()
 }
 
+func (o *Object) NamespacedName() types.NamespacedName {
+	return types.NamespacedName{
+		Namespace: o.GetNamespace(),
```

Comment:
Reading the `types.NamespacedName` doc, it says the `Namespace` is "[mandatory](https://github.com/kubernetes/apimachinery/blob/1cc21f9c83e1ffc69aa85af3bc0cc7467e9e5113/pkg/types/namespacedname.go#L19)", will this be applied to this function? The name "NamespacedName" seems to tell the specific `manifest.Object` must be a namespace scoped resource. 

Another follow up question about the ambiguity of the namespace value. I see this method in this PR is only used in `StatusBuilder`, which namespace value  is very clear (the value can be "default" or "SOME_CUSTOM_NS" for namespace socped and can be empty for cluster scoped resource). But because `manifest.Object` is Public,  users can call it before the manifest is applied, then the empty `namespace` value could be ambiguous between un-assigned namespace scope resource and cluster scoped resource (which both are empty string). This seems to be a risk condition, if it is valid we can at least call it out in the function docstring.

Reply:
Good points.  Added a docstring.  If we can find a better struct or if you have a better name, I'd be very open to one!

We aren't actually using this in many places (yet, though I think the plan was to use it in more places) so we could also just remove it.  I propose we leave it for now and see if it is helpful.